File: apps/nursery_admin_mobile/lib/services/notification_deep_link.dart
/// Admin mobile notification deep links (QA-22 / QA-35).
library;

String? adminNotificationDeepLink(Map<String, dynamic>? data) {
  if (data == null) return null;

  final orderId = data['order_id'];
  final route = data['route']?.toString();

  // Admin Mobile has no returns module (QA-ADM-002). API often sends
  // route=/returns/{id} — never follow that path (QA-35-001).
  if (route != null && route.startsWith('/returns')) {
    if (orderId != null) return '/orders/$orderId';
    return '/orders';
  }

  if (route != null && RegExp(r'^/[A-Za-z0-9/_-]*$').hasMatch(route)) {
    return route;
  }
  if (orderId != null) return '/orders/$orderId';
  if (data['return_id'] != null) {
    return '/orders';
  }
  return null;
}

File: apps/nursery-admin/src/lib/notification-deep-link.ts
/**
 * Admin notification deep links.
 */
export function adminNotificationHref(data: unknown): string | null {
  if (!data || typeof data !== "object") return null;
  const d = data as Record<string, unknown>;
  const route = d.route;
  if (typeof route === "string" && /^\/[A-Za-z0-9/_-]*$/.test(route)) {
    return route.startsWith("/admin") ? route : route;
  }
  if (d.order_id != null && String(d.order_id).match(/^\d+$/)) {
    return `/orders/${d.order_id}`;
  }
  if (d.return_id != null && String(d.return_id).match(/^\d+$/)) {
    return `/returns/${d.return_id}`;
  }
  return null;
}
